docs/concepts/functional-components/macro.md

```diff
@@ -0,0 +1,33 @@
+---
+title: #[functional_component]
+description: The #[functional_component] macro
```

Comment:
I think we should call it "attribute" instead of "macro".
"macro", to me, makes it sound like it would be used like this:
```rust
functional_component! {
  fn my_component() -> Html {
    html! {}
  }
}